File: backend_spring_deprecated/src/main/java/com/voiceiq/backend/feedback/domain/FeedbackReport.java
package com.voiceiq.backend.feedback.domain;

import com.voiceiq.backend.speech.domain.InterviewSession;
import jakarta.persistence.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "feedback_reports")
public class FeedbackReport {

    @Id
    private UUID id;

    @OneToOne(optional = false, fetch = FetchType.LAZY)
    @JoinColumn(name = "session_id")
    private InterviewSession session;

    @Column(name = "overall_score", nullable = false)
    private int overallScore;

    @Column(name = "pace_score", nullable = false)
    private int paceScore;

    @Column(name = "clarity_score", nullable = false)
    private int clarityScore;

    @Column(name = "confidence_score", nullable = false)
    private int confidenceScore;

    @Column(name = "filler_score", nullable = false)
    private int fillerScore;

    @Column(nullable = false, columnDefinition = "text")
    private String summary;

    @Column(nullable = false, columnDefinition = "text")
    private String suggestions;

    @Column(columnDefinition = "text")
    private String strengths;

    @Column(columnDefinition = "text")
    private String weaknesses;

    @Column(name = "better_answer", columnDefinition = "text")
    private String betterAnswer;

    @Column(name = "filler_breakdown", columnDefinition = "text")
    private String fillerBreakdown;

    @Column(name = "hesitation_phrases", columnDefinition = "text")
    private String hesitationPhrases;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @PrePersist
    void onCreate() {
        this.createdAt = LocalDateTime.now();
    }

    public UUID getId() { return id; }
    public void setId(UUID id) { this.id = id; }
    public InterviewSession getSession() { return session; }
    public void setSession(InterviewSession session) { this.session = session; }
    public int getOverallScore() { return overallScore; }
    public void setOverallScore(int overallScore) { this.overallScore = overallScore; }
    public int getPaceScore() { return paceScore; }
    public void setPaceScore(int paceScore) { this.paceScore = paceScore; }
    public int getClarityScore() { return clarityScore; }
    public void setClarityScore(int clarityScore) { this.clarityScore = clarityScore; }
    public int getConfidenceScore() { return confidenceScore; }
    public void setConfidenceScore(int confidenceScore) { this.confidenceScore = confidenceScore; }
    public int getFillerScore() { return fillerScore; }
    public void setFillerScore(int fillerScore) { this.fillerScore = fillerScore; }
    public String getSummary() { return summary; }
    public void setSummary(String summary) { this.summary = summary; }
    public String getSuggestions() { return suggestions; }
    public void setSuggestions(String suggestions) { this.suggestions = suggestions; }
    public String getStrengths() { return strengths; }
    public void setStrengths(String strengths) { this.strengths = strengths; }
    public String getWeaknesses() { return weaknesses; }
    public void setWeaknesses(String weaknesses) { this.weaknesses = weaknesses; }
    public String getBetterAnswer() { return betterAnswer; }
    public void setBetterAnswer(String betterAnswer) { this.betterAnswer = betterAnswer; }
    public String getFillerBreakdown() { return fillerBreakdown; }
    public void setFillerBreakdown(String fillerBreakdown) { this.fillerBreakdown = fillerBreakdown; }
    public String getHesitationPhrases() { return hesitationPhrases; }
    public void setHesitationPhrases(String hesitationPhrases) { this.hesitationPhrases = hesitationPhrases; }
    public LocalDateTime getCreatedAt() { return createdAt; }
}

File: frontend/lib/features/analysis/data/feedback_report.dart
class FeedbackReport {
  const FeedbackReport({
    required this.reportId,
    required this.sessionId,
    required this.overallScore,
    required this.paceScore,
    required this.clarityScore,
    required this.confidenceScore,
    required this.fillerScore,
    required this.transcriptText,
    required this.transcriptHighlights,
    required this.summary,
    required this.strengths,
    required this.weaknesses,
    required this.suggestions,
    required this.betterAnswer,
    required this.fillerBreakdown,
    required this.hesitationPhrases,
  });

  final String reportId;
  final String sessionId;
  final int overallScore;
  final int paceScore;
  final int clarityScore;
  final int confidenceScore;
  final int fillerScore;
  final String transcriptText;
  final List<TranscriptHighlight> transcriptHighlights;
  final String summary;
  final List<String> strengths;
  final List<String> weaknesses;
  final List<String> suggestions;
  final String betterAnswer;
  final List<String> fillerBreakdown;
  final List<String> hesitationPhrases;

  factory FeedbackReport.fromJson(Map<String, dynamic> json) {
    return FeedbackReport(
      reportId: json['reportId'] as String,
      sessionId: json['sessionId'] as String,
      overallScore: json['overallScore'] as int? ?? 0,
      paceScore: json['paceScore'] as int? ?? 0,
      clarityScore: json['clarityScore'] as int? ?? 0,
      confidenceScore: json['confidenceScore'] as int? ?? 0,
      fillerScore: json['fillerScore'] as int? ?? 0,
      transcriptText: json['transcriptText'] as String? ?? '',
      transcriptHighlights:
          (json['transcriptHighlights'] as List<dynamic>? ?? const [])
              .whereType<Map>()
              .map((value) => TranscriptHighlight.fromJson(Map<String, dynamic>.from(value)))
              .toList(),
      summary: json['summary'] as String? ?? '',
      strengths: (json['strengths'] as List<dynamic>? ?? const [])
          .map((value) => value.toString())
          .toList(),
      weaknesses: (json['weaknesses'] as List<dynamic>? ?? const [])
          .map((value) => value.toString())
          .toList(),
      suggestions: (json['suggestions'] as List<dynamic>? ?? const [])
          .map((value) => value.toString())
          .toList(),
      betterAnswer: json['betterAnswer'] as String? ?? '',
      fillerBreakdown: (json['fillerBreakdown'] as List<dynamic>? ?? const [])
          .map((value) => value.toString())
          .toList(),
      hesitationPhrases:
          (json['hesitationPhrases'] as List<dynamic>? ?? const [])
              .map((value) => value.toString())
              .toList(),
    );
  }
}

class TranscriptHighlight {
  const TranscriptHighlight({
    required this.type,
    required this.value,
    required this.startIndex,
    required this.endIndex,
    required this.message,
  });

  final String type;
  final String value;
  final int startIndex;
  final int endIndex;
  final String message;

  factory TranscriptHighlight.fromJson(Map<String, dynamic> json) {
    return TranscriptHighlight(
      type: json['type'] as String? ?? 'FILLER_WORD',
      value: json['value'] as String? ?? '',
      startIndex: json['startIndex'] as int? ?? 0,
      endIndex: json['endIndex'] as int? ?? 0,
      message: json['message'] as String? ?? '',
    );
  }
}
